br/pkg/lightning/errormanager/errormanager.go

```diff
@@ -504,8 +531,144 @@ func (em *ErrorManager) ResolveAllConflictKeys(
 	return errors.Trace(g.Wait())
 }
 
+// ReplaceConflictKeys query all conflicting rows (handle and their
+// values) from the current error report and resolve them concurrently
+// by replacing the necessary rows and reserving the others.
+func (em *ErrorManager) ReplaceConflictKeys(
+	ctx context.Context,
+	tbl tidbtbl.Table,
+	tableName string,
+	pool *utils.WorkerPool,
+	decoder *kv.TableKVDecoder,
+	fnGetLatest func(ctx context.Context, key []byte) ([]byte, error),
+	fnDeleteKey func(ctx context.Context, key []byte) error,
+) error {
+	if em.db == nil {
+		return nil
+	}
+
+	sessionOpts := encode.SessionOptions{
+		// TODO: need to find the correct value for SQLMode
+		SQLMode: mysql.ModeStrictAllTables,
+	}
+	encoder, err := kv.NewBaseKVEncoder(&encode.EncodingConfig{
+		Table:          tbl,
+		SessionOptions: sessionOpts,
+		Logger:         em.logger,
+	})
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	g, gCtx := errgroup.WithContext(ctx)
+	pool.ApplyOnErrorGroup(g, func() error {
+		// check index KV first
+		rawKeyRows, err := em.db.QueryContext(
+			gCtx, fmt.Sprintf(selectIndexConflictKeysReplace, em.schemaEscaped),
+			tableName)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		for rawKeyRows.Next() {
+			var rawKey []byte
+			if err := rawKeyRows.Scan(&rawKey); err != nil {
+				return errors.Trace(err)
+			}
+			em.logger.Debug("got rawKey from table",
+				zap.Binary("rawKey", rawKey))
+
+			value, err := fnGetLatest(gCtx, rawKey)
+			em.logger.Debug("got value from fnGetLatest",
+				zap.Binary("rawKey", rawKey),
+				zap.Binary("value", value))
+			if err != nil {
+				return errors.Trace(err)
+			}
+
+			rows, err := em.db.QueryContext(
```

Comment:
currently we use two SQL that query "raw_key" first and then query its "raw_value" and "raw_handle". Can we use one SQL to do it? like SELECT ... ORDER BY raw_key.